lib/puppet/type/image_streamer_plan_script.rb

```diff
@@ -0,0 +1,44 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+Puppet::Type.newtype(:image_streamer_plan_script) do
+  desc "Image Streamer's Plan Script"
+
+  # :nocov:
+  ensurable do
+    defaultvalues
+
+    newvalue(:found) do
+      provider.found
+    end
+
+    newvalue(:retrieve_differences) do
+      provider.retrieve_differences
+    end
+  end
+  # :nocov:
+
+  newparam(:name, namevar: true) do
+    desc 'Plan Script name'
+  end
+
+  newparam(:data) do
+    desc 'Plan Script data hash containing all specifications for the system'
+    validate do |value|
+      raise 'Inserted value for data is not valid' unless value.class == Hash
```

Comment:
The mock `modified_config[:data] = ' '` is a string, and strings actually do respond to [] and []= so that would not raise an error and is according to expected. 
To fix the issue, use something like `modified_config[:data] = 5`, since `int` does not respond to [] nor []= and is one of the use cases in which we want to fail early.